contracts/test/eip155.ts

```diff
@@ -36,8 +74,52 @@ describe('EIP-155', function () {
     let receipt = await testContract.provider.waitForTransaction(
       plainResp.hash,
     );
-    expect(receipt.logs[0].data).equal(
-      '0xfedcba9876543210fedcba9876543210fedcba9876543210fedcba9876543210',
-    );
+    expect(plainResp.data).eq(calldata);
+    expect(receipt.logs[0].data).equal(EXPECTED_EVENT);
+  });
+
+  it('Other-Signed transaction submission via wrapped provider', async function () {
+    const provider = testContract.provider;
+    const signedTx = await testContract.sign({
+      nonce: await provider.getTransactionCount(
+        await testContract.publicAddr(),
+      ),
+      gasPrice: await provider.getGasPrice(),
+      gasLimit: 250000,
+      to: testContract.address,
+      value: 0,
+      data: '0x',
+      chainId: 0,
+    });
+
+    let plainResp = await provider.sendTransaction(signedTx);
+    let receipt = await provider.waitForTransaction(plainResp.hash);
+    expect(plainResp.data).eq(calldata);
+    expect(receipt.logs[0].data).equal(EXPECTED_EVENT);
+  });
+
+  it('Self-Signed transaction submission via wrapped wallet', async function () {
+    const provider = testContract.provider;
+    const wallet = sapphire.wrap(getWallet(0).connect(provider));
+
+    const signedTx = await wallet.signTransaction({
+      gasLimit: 250000,
+      to: testContract.address,
+      value: 0,
+      data: calldata,
+      chainId: (await provider.getNetwork()).chainId,
+      gasPrice: await provider.getGasPrice(),
+      nonce: await provider.getTransactionCount(wallet.address),
+    });
+
+    // Calldata should be encrypted when we wrap the wallet provider
+    let x = await provider.sendTransaction(signedTx);
+    expect(entropy(x.data)).gte(3.8);
+    expect(x.data).not.eq(calldata);
+
+    let r = await provider.waitForTransaction(x.hash);
+    expect(r.logs[0].data).equal(EXPECTED_EVENT);
   });
+
+  // TODO: test error conditions?
```

Comment:
👍

Reply:
This is being handled in another PR: https://github.com/oasisprotocol/sapphire-paratime

I'm going to leave the TODO: in for now